### DISCUSSION TITLE
functionnality Traits in Adonis V5

### DISCUSSION THREAD

**Discussion by sergefabre** [adonisjs/core#2967] [Help] [answered]
Hello

In my project on Adonis V4.1, i use the functionnality "Traits", 
For example, i have a trait "UpSert" that i could invoke from all my models 

In Adonis V5, i dont find this functionnality.

Is there a similar functionnality in V5 ?

Ultimately, I would like use this : https://www.postgresqltutorial.com/postgresql-upsert/

Thanks for the help

**Comment by arthur-er** [comment, marked as answer]
There's a feature similar to traits based in inheritance and composition, its implemented here:
https://github.com/lookinlab/adonis-lucid-filter/blob/develop/src/Mixin/index.ts

Also in Adonis docs there is the compose helper: https://docs.adonisjs.com/guides/helpers#compose

**Comment by thetutlage** [comment]
Traits cannot fundamentally work with TypeScript. As @arthur-er said, please refer to the mixin docs

**Reply by sergefabre** [reply]
thanks a lot @thetutlage and @arthur-er
